pkg/minikube/constants/constants.go

```diff
@@ -32,13 +32,12 @@ var (
 
 const (
 	// DefaultKubernetesVersion is the default Kubernetes version
-	// dont update till #10545 is solved
 	DefaultKubernetesVersion = "v1.23.5"
 	// NewestKubernetesVersion is the newest Kubernetes version to test against
 	// NOTE: You may need to update coreDNS & etcd versions in pkg/minikube/bootstrapper/images/images.go
-	NewestKubernetesVersion = "v1.23.6-rc.0"
```

Comment:
any particular reason we're changing this here?

Reply:
hi, @sharifelgamal, yes, the tag "v1.23.6-rc.0" was never released  and it should not be used (but was picked up by the old code logic now corrected in this pr); more details are in the issue's description: https://github.com/kubernetes/minikube/issues/13776

Reply:
@sharifelgamal in the current master, we still use the unreleased "v1.23.6-rc.0" tag for the NewestKubernetesVersion, but i can remove this change from this pr if needed but then we should still correct that in a separate one?

Reply:
we're upgrading newest to 1.24.0 in a separate PR soon, so this is kind of a moot point. but in general, we allow RCs and betas to be NewestKuberenetesVersion, mostly for testing.

Reply:
yes, we do allow RCs and betas, that's fine, but at times we've also (mistakenly) picked up some _tags_ that were not _released_ tags - ie, that never made to the https://github.com/kubernetes/kubernetes/releases list

the [v1.23.6-rc.0](https://github.com/kubernetes/kubernetes/releases/tag/v1.23.6-rc.0) here is a good example of that, while eg [v1.24.0-rc.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.24.0-rc.1) is an example of a valid, ie - _released_ RC tag

this pr should fix that issue so that we only pick up _released_ tags (including RCs, betas, etc.)